internal/manager/setup.go

```diff
@@ -192,51 +192,88 @@ func setupAdmissionServer(
 	return nil
 }
 
-func setupDataplaneAddressFinder(ctx context.Context, mgrc client.Client, c *Config) (*dataplane.AddressFinder, error) {
+// setupDataplaneAddressFinder returns a default and UDP address finder. These finders return the override addresses if
+// set or the publish service addresses if no overrides are set. If no UDP overrides or UDP publish service are set,
+// the UDP finder will also return the default addresses. If no override or publish service is set, this function
+// returns nil finders and an error.
+func setupDataplaneAddressFinder(
+	ctx context.Context,
```

Comment:
It just occurred to me that this should not accept `context.Context`: the address getter should use a context that's passed to it when invoked not when creating it.

This way we'll be able to pass down the context from e.g. reconcile request as in https://github.com/Kong/kubernetes-ingress-controller/blob/3db70101212dde5733cca2a13d87634f4a696af2/internal/controllers/configuration/zz_generated_controllers.go#L404

We can do that in a separate PR though.

Reply:
Eh, makes sense, may as well try and do it here a81403e58df2c037964567f745e7bdc2ee73baee